todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksDaoTest.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2017, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp.data.source.local
+
+import android.arch.persistence.room.Room
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import com.example.android.architecture.blueprints.todoapp.data.Task
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.notNullValue
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class) class TasksDaoTest {
+
+    private lateinit var database: ToDoDatabase
+
+    @Before fun initDb() {
+        // using an in-memory database because the information stored here disappears when the
+        // process is killed
+        database = Room.inMemoryDatabaseBuilder(InstrumentationRegistry.getContext(),
+                ToDoDatabase::class.java).build()
+    }
+
+    @After fun closeDb() {
+        database.close()
+    }
+
+    @Test fun insertTaskAndGetById() {
+        // When inserting a task
+        database.taskDao().insertTask(TASK)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById(TASK.id)
+
+        // The loaded data contains the expected values
+        assertTask(loaded, "id", "title", "description", true)
+    }
+
+    @Test fun insertTaskReplacesOnConflict() {
+        //Given that a task is inserted
+        database.taskDao().insertTask(TASK)
+
+        // When a task with the same id is inserted
+        val newTask = Task("title2", "description2", "id").apply {
+            isCompleted = true
+        }
+        database.taskDao().insertTask(newTask)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById(TASK.id)
+
+        // The loaded data contains the expected values
+        assertTask(loaded, "id", "title2", "description2", true)
```

Comment:
Use constants for these strings

Reply:
Done (cherry-picked dev-todo-mvp-kotlin modifications).